sphinx/ext/autosummary/generate.py

```diff
@@ -514,13 +516,15 @@ def generate_autosummary_docs(
     if base_path is not None:
         sources = [os.path.join(base_path, filename) for filename in sources]
 
+    assert app is not None
```

Comment:
We have an `if app` guard afterwards so I'm not sure if the `app = None` is allowed only for tests or if there was a real reason not to have any `app`.

Reply:
just did it to satisfy type checking, but yeh if you look at `AutosummaryRenderer.__init__` an `app` is definitely required

Reply:
I removed the type guard